File: Natixis Craftsmanship Practices/Breaking Dependencies/MarketTrading/java/src/main/java/com/markettrading/external/XmlObjectMapper.java
package com.markettrading.external;

@DoNotModify
public class XmlObjectMapper {

    public <T> T read(String xml, Class<T> clazz) throws ExternalSideEffectException {
        throw new ExternalSideEffectException("Native XML deserialization forbidden in test");
    }

    public String write(Object obj) throws ExternalSideEffectException {
        throw new ExternalSideEffectException("Native XML serialization forbidden in test");
    }
}

File: Natixis Craftsmanship Practices/Breaking Dependencies/MarketTrading/csharp/External/XmlObjectMapper.cs
namespace MarketTrading.External;

[DotNotModify]
public class XmlObjectMapper
{
    public T Read<T>(string xml,  Type o) {
        throw new ExternalSideEffectException("Native XML deserialization forbidden in test");
}

    public string Write(object obj){
        throw new ExternalSideEffectException("Native XML serialization forbidden in test");
    }
    
}